### PR TITLE
TH2-1517 Use sse search api to render events

### PR DESCRIPTION
**Pull request by VekesserVladimir**

### CODE REVIEW

**Comment by rzvdv** [comment]
Switching to flat view breaks app
Initial loading with event path doesn't expand events

File: src/stores/events/EventsStore.ts [hunk 1]
```diff
@@ -181,41 +185,115 @@ export default class EventsStore {
...
+		return new Promise<void>((res, rej) => {
+			this.eventTreeStatusCode = null;
+			this.selectedNode = null;
+			this.isLoadingRootEvents = true;
+			this.eventTree = [];
 
-		this.selectedNode = null;
-		this.isLoadingRootEvents = true;
+			try {
+				const eventSource = this.api.sse.getEventSource({
```
Comment (rzvdv) [inline src/stores/events/EventsStore.ts:197]: sse channel should be member of eventstore and it should be close upon applying filter etc

File: src/stores/events/EventsStore.ts [hunk 2]
```diff
@@ -181,41 +185,115 @@ export default class EventsStore {
...
-	private eventTreeAC: AbortController | null = null;
-
 	@action
-	private fetchEventTree = async () => {
-		if (this.eventTreeAC) {
-			this.eventTreeAC.abort();
-		}
-		this.eventTreeAC = new AbortController();
-		this.eventTreeStatusCode = null;
+	private fetchEventTree = () => {
```
Comment (rzvdv) [inline src/stores/events/EventsStore.ts:189]: store is getting quite large, maybe we can create some EventsDataStore to move everything related to fetching there?

File: src/stores/events/EventsStore.ts [hunk 3]
```diff
@@ -181,41 +185,115 @@ export default class EventsStore {
...
+
+	@action
+	private async loadRootEvents() {
+		const loadEventById = async (eventId: string, childList: EventTreeNode[]) => {
+			try {
+				const event = await this.api.events.getEvent(eventId);
+				if (event.parentEventId === null) {
+					if (this.eventTree.find(node => node.eventId === event.eventId)) return;
+
+					const rootNode = convertEventActionToEventTreeNode(event);
```
Comment (rzvdv) [inline src/stores/events/EventsStore.ts:276]: Maybe converting is unnecessary ? if we receive EventAction let's make it a tree of EventActions?

**Comment by YourWeebyNeighbor** [comment]
Let's not merge it just yet. We need to test how this change works on large amount of events

**Review by YourWeebyNeighbor — APPROVED** [review]

**Pull request merged**